### ISSUE TITLE
thread 'tokio-runtime-worker' panicked

### ISSUE THREAD

**Issue by testerez** [prisma/prisma#16268]
### Bug description

I'm getting this error when running my tests. Those tests are actually writing to the DB. It's hard to know exactly what is causing the error since it's not consistent and it does not alway happen on the same test.

Here is the error backtrace:

```
thread 'tokio-runtime-worker' panicked at 'Expected weak reference to be valid.', /Users/runner/work/prisma-engines/prisma-engines/query-engine/schema/src/lib.rs:44:24
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic_display
   3: core::panicking::panic_str
   4: core::option::expect_failed
   5: query_core::response_ir::internal::serialize_internal
   6: query_core::response_ir::ir_serializer::IrSerializer::serialize
   7: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  10: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  11: <tokio::task::task_local::TaskLocalFuture<T,F> as core::future::future::Future>::poll
  12: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  13: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  14: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  15: <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll
  16: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  17: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  18: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  19: tokio::runtime::task::core::CoreStage<T>::poll
  20: tokio::runtime::task::harness::Harness<T,S>::poll
  21: std::thread::local::LocalKey<T>::with
  22: tokio::runtime::scheduler::multi_thread::worker::Context::run_task
  23: tokio::runtime::scheduler::multi_thread::worker::Context::run
  24: tokio::macros::scoped_tls::ScopedKey<T>::set
  25: tokio::runtime::scheduler::multi_thread::worker::run
  26: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
  27: tokio::runtime::task::harness::Harness<T,S>::poll
  28: tokio::runtime::blocking::pool::Inner::run```
```

### How to reproduce

So far I don't have a sharable way to reproduce.

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: v16.15.1

### Prisma Version

```
prisma@4.6.1
```

**Comment by SevInf** [comment]
Hi @testerez!
Unfortunately, we'd need a better way to reproduce the issue in order to be able to investigate it. Could you maybe share a schema? And even though you say it does not happen consistently, maybe you'd be able to share a couple of queries where it is known to happen from time to time?

**Comment by Druue** [comment]
Hey @testerez 

We unfortunately can't reproduce your issue without further information. Could you please get back to us with the information that @SevInf mentioned so that we can try fixing your issue?

Thank you 🙏

**Comment by testerez** [comment]
Hey @Druue , I realized I had this error when I was running multiple queries in parallel with `Promise.all()` and one of the queries was failing. In that case, the original error was not surfacing. I removed some `Promise.all()` in my tests and I'm not getting those errors anymore.

**Comment by janpio** [comment]
Is this still reproducible with `Promise.all()` @testerez? If so, we would appreciate a new issue with a reproduction project so we can experience that ourselves. Although great you have a workaround, this should of course still not break.

**Comment by Malewooo16** [comment]
I am not sure but when i run this code in node it works but in nextjs server actions it doesn't 
**Function**
`"use server";
import prisma from "@/db/prisma";
export async function updateStudentResults(
  students,
): Promise<{ success: boolean; message: string }> {
  try {
    await prisma.$transaction(async (tx) => {
      const updates = students.map(async (studentData) => {
        const existingStudent = await tx.studentT.findUnique({
          where: { id: studentData.id },
        });

        if (!existingStudent) {
          throw new Error(`Student with id ${studentData.id} does not exist.`);
        }

        let updatedResults = [];

        if (existingStudent.results) {
          updatedResults = existingStudent.results.map((result) => {
            if (result.name === studentData.courseName) {
              return {
                name: studentData.courseName,
                scores: studentData.newScores,
                rank: studentData.rank,
                avg: studentData.avg,
              };
            }
            return result;
          });

          const courseExists = updatedResults.some(
            (result) => result.name === studentData.courseName,
          );
          if (!courseExists) {
            updatedResults.push({
              name: studentData.courseName,
              scores: studentData.newScores,
              rank: studentData.rank,
              avg: studentData.avg,
            });
          }
        } else {
          updatedResults.push({
            name: studentData.courseName,
            scores: studentData.newScores,
            rank: studentData.rank,
            avg: studentData.avg,
          });
        }

        return tx.studentT.update({
          where: { id: studentData.id },
          data: { results: updatedResults },
        });
      });

      // Ensure all updates are executed
      await Promise.all(updates);
    });

    return { success: true, message: "Student results updated successfully" };
  } catch (error) {
    console.error("Error updating student results:", error);
    return { success: false, message: `Error updating student results` };
  } finally {
    await prisma.$disconnect();
  }
}
`
**Schema**
model StudentT {
  id                Int                @id @default(autoincrement())
  name              String
  schoolId          Int
  classId           Int?
  admissionId       String?
  parentId          Int?
  courseEnrollments CourseEnrollment[]
  results            Json[]
  remarks           StudentRemarks[]
  showRecords       Boolean  @default(false)
  studentData       Admission?          @relation(fields: [admissionId], references: [id])
  class             Classes?           @relation(fields: [classId], references: [id])
  parent            Parent?            @relation(fields: [parentId], references: [id])
  school            School             @relation(fields: [schoolId], references: [id])
}
